### ISSUE TITLE
Automatically register SpecFlow bindings

### ISSUE THREAD

**Issue by mbhoek** [solidtoken/SpecFlow.DependencyInjection#7]
Currently the developer has to write a little bit of setup code to setup the DI for SpecFlow (as described in the [README.md](/solidtoken/SpecFlow.DependencyInjection/blob/master/README.md)). This setup code needs to register all the SpecFlow ``` [Binding]```s otherwise the SpecFlow tests will fail with a obscure error message.

As a developer, I want to automatically register all ```[Binding]```s in the same Assembly where the ```[ScenarioDependencies]``` is defined.

- [x] Add a ```AutoRegisterBindings``` (default True) to ```[ScenarioDependencies```
- [x] Automatically register the appropriate ```[Binding]```s

**Comment by mbhoek** [comment]
This would theoretically be a breaking change (if we make True the default).

**Comment by alenmeister** [comment]
Any updates on this?

**Comment by mbhoek** [comment]
@alenmeister Thanks for the reminder. I have no urgent need for this so I've not looked at it yet. 
I am contemplating to use 'configuration by convention' (like the `ConfigureServices` method in the `Startup` class in ASP.NET Core) instead of an AttributeFlag like it is now.

Is there anything in particular you would want from this feature?

**Comment by alenmeister** [comment]
@mbhoek Not sure if I quite understood what you meant. Wouldn't a simple extension method do the trick? It'd be nice if one could simply do services.AutoRegisterBindings as you pointed out in the description. But hey, I'm not proficient in C# as I've only been working with the language for three weeks.

**Comment by 304NotModified** [comment]
> otherwise the SpecFlow tests will fail with a obscure error message.

To be clear, that's this message:

```
System.NullReferenceException: Object reference not set to an instance of an object.
    at lambda_method(Closure , IContextManager , String )
   at TechTalk.SpecFlow.Bindings.BindingInvoker.InvokeBinding(IBinding binding, IContextManager contextManager, Object[] arguments, ITestTracer testTracer, TimeSpan& duration) in D:\a\1\s\TechTalk.SpecFlow\Bindings\BindingInvoker.cs:line 69
   at TechTalk.SpecFlow.Infrastructure.TestExecutionEngine.ExecuteStepMatch(BindingMatch match, Object[] arguments) in D:\a\1\s\TechTalk.SpecFlow\Infrastructure\TestExecutionEngine.cs:line 501
   at TechTalk.SpecFlow.Infrastructure.TestExecutionEngine.ExecuteStep(IContextManager contextManager, StepInstance stepInstance) in D:\a\1\s\TechTalk.SpecFlow\Infrastructure\TestExecutionEngine.cs:line 422
   at TechTalk.SpecFlow.Infrastructure.TestExecutionEngine.Step(StepDefinitionKeyword stepDefinitionKeyword, String keyword, String text, String multilineTextArg, Table tableArg) in D:\a\1\s\TechTalk.SpecFlow\Infrastructure\TestExecutionEngine.cs:line 581
   at TechTalk.SpecFlow.TestRunner.Given(String text, String multilineTextArg, Table tableArg, String keyword) in D:\a\1\s\TechTalk.SpecFlow\TestRunner.cs:line 80
   at 
```

**Comment by 304NotModified** [comment]
This will be fixed in #25
